pkg/ddl/tests/metadatalock/mdl_test.go

```diff
@@ -935,7 +935,7 @@ func TestMDLPreparePlanCacheExecute(t *testing.T) {
 	tk.MustQuery("select * from t2")
 	tk.MustExec(`set @a = 2, @b=4;`)
 	tk.MustExec(`execute stmt_test_1 using @a, @b;`)
-	tk.MustQuery("select @@last_plan_from_cache;").Check(testkit.Rows("1"))
+	tk.MustQuery("select @@last_plan_from_cache;").Check(testkit.Rows("0"))
```

Comment:
Should we also have a separated test case for the case about preparing `select ... for update` statement outside transaction? This test targets on testing MDL, and I'm afraid that this test might being modified / removed without being known by us and making the case lose coverage...

Reply:
I enhancemented this test, PTAL